### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by iamargus95**
1. Changed githubapi.UserData and githubapi.RepoData to struct methods.
2. Added Functionality to take multiple usernames from CLI. 
3. Added concurrency
4. Added makefile and build instructions in README.md

### CODE REVIEW

**Comment by iamargus95** [comment]
Added ability to parse usernames directly from the CLI.

File: githubapi/githubapi.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"strconv"
 )
 
+const UrlGetRequest = "https://api.github.com/users/"
```
Comment (jdk2588) [inline githubapi/githubapi.go]: Constant are in CAPS https://golang.org/doc/effective_go#constants

File: githubapi/githubapi.go [hunk 2]
```diff
@@ -41,7 +43,7 @@ func responseToRepoData(data []byte) ReposInfoArray {
 
 func GetUserData(username string) Userinfo {
 
-	url := "https://api.github.com/users/" + username
+	url := UrlGetRequest + username
```
Comment (jdk2588) [inline githubapi/githubapi.go]: Please use url constructor library, like https://pkg.go.dev/net/url

File: githubapi/githubapi.go [hunk 3]
```diff
@@ -71,7 +73,7 @@ func GetReposData(username string, noOfRepos int) ReposInfoArray {
 
 	for i := 1; i <= ((noOfRepos / 100) + 1); i++ {
 
-		url := "http://api.github.com/users/" + username + "/repos?type=public&per_page=100&page=" + strconv.Itoa(i)
+		url := UrlGetRequest + username + "/repos?type=public&per_page=100&page=" + strconv.Itoa(i)
```
Comment (jdk2588) [inline githubapi/githubapi.go]: Please use url constructor library, like https://pkg.go.dev/net/url

File: main.go [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
-	repodata := githubapi.RepoData(reposDetails)
-	io.WriteToFile(username, strings.Split(userdata, ","))
-	io.WriteToFile(username, repodata)
+	var wg sync.WaitGroup
+	dataToFile := make(chan []string)
+
+	for _, username := range usernames {
+		wg.Add(1)
+		go worker(username, dataToFile, &wg)
+		io.WriteToFile(username, dataToFile)
```
Comment (jdk2588) [inline main.go]: Why write to file is outside of Go routine?

File: io/writeToFile.go [hunk 5]
```diff
@@ -6,7 +6,9 @@ import (
 	"os"
 )
 
-func WriteToFile(username string, data []string) {
+func WriteToFile(username string, dataToFile chan []string) error {
```
Comment (jdk2588) [inline io/writeToFile.go]: Write to File should only care about content like string data, should not use any channel.

**Review by jdk2588 — CHANGES_REQUESTED** [review]

**Comment by iamargus95** [comment]
Fixed Concurrency issues where data was being written to random files depending on which api call gives a faster response. Workaround :- Used a channel of map[string][]string to overcome it.

Fixed waitgroups not working as expected.  [ defer wg.Done() would not work inside the fetch worker func. ] 
Workaround :- Did a wg.Add(len(usernames)) before the write worker func.

Added Benckmarking tests and updated the README.

**Pull request merged**